dialoptions.go

```diff
@@ -187,6 +189,14 @@ func WithDefaultCallOptions(cos ...CallOption) DialOption {
 	})
 }
 
+// withDisableGlobalOptions returns a DialOption which sets the
+// disableGlobalOptions bool to true.
```

Comment:
This comment isn't very helpful.

But neither is `// withDisableGlobalOptions disables global options` :)

Maybe:

```go
// withDisableGlobalOptions [returns a DialOption that?] prevents the ClientConn from applying
// the global DialOptions (set via AddGlobalDialOptions).
```

?